### PR TITLE
Document Session and Segment

### PR DESCRIPTION
**Pull request by tzerrell**

### CODE REVIEW

File: risc0/zkvm/src/receipt.rs [hunk 1]
```diff
@@ -384,7 +393,12 @@ pub struct SegmentReceipt {
 pub struct ReceiptMetadata {}
 
 impl SessionReceipt {
-    /// TODO
+    /// Verifies the integrity of this receipt.
```
Comment (pdg744) [inline risc0/zkvm/src/receipt.rs:396]: should this say "of this session" ?

File: risc0/zkvm/src/receipt.rs [hunk 2]
```diff
@@ -362,20 +362,29 @@ impl Global {
...
+    /// The constituent [SegmentReceipt]s.
+    ///
+    /// Together these can be used by [SessionReceipt::verify] to
+    /// cryptographically prove that this full Session was faithfully executed.
     pub segments: Vec<SegmentReceipt>,
 
-    /// TODO
+    /// The public data written by the guest in this Session.
+    ///
+    /// This data is cryptographically attested to in [SessionReceipt::verify].
```
Comment (pdg744) [inline risc0/zkvm/src/receipt.rs:376]: "attested to in `::verify` " feels off to me.

Perhaps: "This data is cryptographically authenticated in [SessionReceipt::verify]."

File: risc0/zkvm/src/receipt.rs [hunk 3]
```diff
@@ -384,7 +393,12 @@ pub struct SegmentReceipt {
 pub struct ReceiptMetadata {}
 
 impl SessionReceipt {
-    /// TODO
+    /// Verifies the integrity of this receipt.
+    ///
+    /// Uses the ZKP system to cryptographically verify that each constituent
```
Comment (pdg744) [inline risc0/zkvm/src/receipt.rs:398]: nice!

**Comment by tzerrell** [comment]
Re-upping this now that #503 has landed -- is this ready to land?

**Review by flaub — APPROVED** [review]
LGTM

**Pull request merged**